akka-actor-typed/src/main/scala/akka/actor/typed/internal/ActorContextImpl.scala

```diff
@@ -274,7 +276,7 @@ import scala.util.Success
   def pipeToSelf[Value](
       future: CompletionStage[Value],
       applyToResult: akka.japi.function.Function2[Value, Throwable, T]): Unit = {
-    future.whenComplete { (value, ex) =>
+    future.handle { (value, ex) =>
```

Comment:
> When the CompletionStage is failed, handle performant better.

Isn't `whenComplete` more semantically what we are using it for? A callback, not a transformation. For Java API docs of whenComplete: "Unlike method handle, this method is not designed to translate completion outcomes"

What is more performant? Is that due to some implementation detail in Java or what do you base that on? Do we care about the performance of the exception case?

Reply:
Not obvious from just looking at the impl in JDK 17, they look close to identical.

Reply:
https://github.com/scala/scala/pull/10378 waiting